core/src/main/java/aima/core/search/api/DefinedGoalStatesProblem.java

```diff
@@ -0,0 +1,11 @@
+package aima.core.search.api;
+
+import java.util.List;
+
+/**
+ * we need a defined goal states for bidirectional problems
+ * @author wormi
+ */
+public interface DefinedGoalStatesProblem<A, S> extends Problem<A, S> {
```

Comment:
I think the approach taken on the aima3e branch is a little clearer, see:

https://github.com/aimacode/aima-java/blob/AIMA3e/aima-core/src/main/java/aima/core/search/framework/problem/BidirectionalProblem.java

this lets you specify both starting points cleanly and to configure the sub-searches differently if needed. Have you reviewed aima3e's corresponding implementation and unit tests?